macros.in

```diff
@@ -380,10 +380,11 @@ Supplements:   (%{name} = %{version}-%{release} and langpacks-%{1})\
 #		"w3.zstdio"	zstd level 3, zstd's default
 #		"w19T8.zstdio"	zstd level 19 using 8 threads
 #		"w7T0.zstdio"	zstd level 7 using %{getncpus} threads
+#		"w19.zstdio"	zstd level 19 (v6 default)
 #		"w.ufdio"	uncompressed
 #
 #%_source_payload	w9.gzdio
-#%_binary_payload	w9.gzdio
+%_binary_payload	%[ %_rpmformat >=6 ? "w19.zstdio" : "w9.gzdio" ]
```

Comment:
Why make it conditional?  All versions of Fedora (+ RHEL 9 and derivatives) have been building (binary) packages with zstd payloads for years at this point.  As far as I can tell just about every existing supported distribution has a version of RPM that supports zstd and builds RPM with zstd support enabled.  EL8+, SLES 15, etc.

The list of distros that would choke on a zstd package is... EL7 Extended Lifecycle, and maybe SLES 12 LTSS?  

I would think it would be OK to use zstd even for source packages at this point.

Reply:
We try to represent a relatively independent upstream. For v4 packages, the default payload compression has been gzip for the ~20 years of its existence, we don't want to change that *now*.